lkcrypto.c

```diff
@@ -171,6 +184,42 @@ static int enc_dec(struct cipher_tfm_t *tfm, const u8 *key, size_t key_len,
 	return ret;
 }
 
+static int enc_dec_aead(struct cipher_aead_tfm_t *tfm, const u8 *key,
+			size_t key_len, u8 *iv, size_t iv_len, u8 *auth,
+			size_t auth_len, u8 *aad, size_t aad_len,
+			u8 *input_buffer, size_t input_buffer_len, int enc_dec)
+{
+	int ret = 0;
+	struct scatterlist sg[3];
+	ret = crypto_aead_setkey(tfm->tfm, key, key_len);
+	if (ret) {
+		return ret;
+	}
+	ret = crypto_aead_setauthsize(tfm->tfm, auth_len);
+	if (ret) {
+		return ret;
+	}
+	sg_init_table(sg, 3);
```

Comment:
Does this need a corresponding `sg_free_table` or something?

Reply:
I don't think so. `sg_free_table` is needed when we call `sg_alloc_table`. Here we use `sg_init_table` that is by my understanding just setting pointers to the correct offsets inside stack allocated `scatterlist sg` ? i.e. all memory is allocated elsewhere scatterlist just keep pointers to that memory.